3737/1/hash_of_gems.rb

```diff
@@ -0,0 +1,41 @@
+require_relative 'gem_info.rb'
+
+# class that collect all gems with their info
+class HashGems
+  def initialize
+    @gems = {}
+  end
+
+  def gems_list(option)
+    file = YAML.load_file(option[:file])
+    hash_of_gems(file['gems'])
+    select_options(option)
+  end
+
+  private
+
+  def hash_of_gems(file)
+    file.each do |lib|
+      stat = GemInfo.new(lib).getgem_stat
+      next unless stat
+      @gems[lib] = stat
+    end
+    @gems
+  end
+
+  def select_options(opt)
+    same_gem_name(opt)
+    show_top_gems(opt)
+  end
+
+  def same_gem_name(opt)
```

Comment:
maybe `gems_with_similar_names` better reflect what is happenning in this method?